src/ast/node/implement.rs

```diff
@@ -10,9 +10,16 @@ use rustc_hash::FxHashSet;
 
 #[node(comment)]
 pub struct ImplNode {
+    /// generics stands for the generics types when implements for a structure
```

Comment:
not only structure, generic defined here can be used in all methods in block even if the implement target type is not a generic type